include/seqcmd.h

```diff
@@ -0,0 +1,510 @@
+} TempoCmdType;
+
+// Subset of `SEQCMD_OP_SETUP_CMD`
+typedef enum {
+    /* 0x0 */ SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME,
+    /* 0x1 */ SEQCMD_SUB_OP_SETUP_SEQ_UNQUEUE,
+    /* 0x2 */ SEQCMD_SUB_OP_SETUP_RESTART_SEQ,
+    /* 0x3 */ SEQCMD_SUB_OP_SETUP_TEMPO_SCALE,
+    /* 0x4 */ SEQCMD_SUB_OP_SETUP_TEMPO_RESET,
+    /* 0x5 */ SEQCMD_SUB_OP_SETUP_PLAY_SEQ,
+    /* 0x6 */ SEQCMD_SUB_OP_SETUP_SET_FADE_TIMER,
+    /* 0x7 */ SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME_IF_QUEUED,
+    /* 0x8 */ SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME_WITH_SCALE,
+    /* 0x9 */ SEQCMD_SUB_OP_SETUP_SET_CHANNEL_DISABLE_MASK,
+    /* 0xA */ SEQCMD_SUB_OP_SETUP_SET_PLAYER_FREQ,
+    /* 0xE */ SEQCMD_SUB_OP_SETUP_POP_CACHE = 0xE,
+    /* 0xF */ SEQCMD_SUB_OP_SETUP_RESET_SETUP_CMDS
+} SetupCmdType;
+
+// Subset of `SEQCMD_OP_GLOBAL_CMD`
+typedef enum {
+    /* 0x0 */ SEQCMD_SUB_OP_GLOBAL_SET_SOUND_MODE,
+    /* 0x1 */ SEQCMD_SUB_OP_GLOBAL_DISABLE_NEW_SEQUENCES
+} SubCmdType;
+
+// ==== Audio Sequence Primary Commands ====
+
+/**
+ * Play a sequence on a given seqPlayer
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to play the sequence
+ * @param fadeInDuration the sequence will gradually fade in over a duration
+ * @param seqArg no effect: `< 0x7F`, skip ticks: `= 0x7F`, will not play: `>= 0x80` (see note)
+ * @param seqId the id of the sequence to play, see NA_BGM_XXX
+ *
+ * @note seqArg will also be stored in gActiveSeqs.seqId, any check against that seqId must also include seqArg.
+ * seqArg = 0x7F  will interpret the duration as the number of ticks to skip.
+ * seqArg >= 0x80 was intented to load a soundFont asynchronously but the code is unfinished (based on MM).
+ */
+#define SEQCMD_PLAY_SEQUENCE(seqPlayerIndex, fadeInDuration, seqArg, seqId)                                           \
+    Audio_QueueSeqCmd((SEQCMD_OP_PLAY_SEQUENCE << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(fadeInDuration) << 16) | \
+                      ((u8)(seqArg) << 8) | (u16)(seqId))
+
+/**
+ * Stop a sequence on a given seqPlayer
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to play the sequence
+ * @param fadeOutDuration the sequence will gradually fade out over a duration
+ *
+ * @note the 0xFF in the command is not read from at all, but is common in all Stop SeqPlayer Commands
+ */
+#define SEQCMD_STOP_SEQUENCE(seqPlayerIndex, fadeOutDuration)                                 \
+    Audio_QueueSeqCmd((SEQCMD_OP_STOP_SEQUENCE << 28) | 0xFF | ((u8)(seqPlayerIndex) << 24) | \
+                      ((u8)(fadeOutDuration) << 16))
+
+/**
+ * Add a sequence to a queue of sequences associated with a given seqPlayer.
+ * If the sequence is first in queue, play the sequence
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to queue the sequence
+ * @param fadeInDuration if the sequence is played, gradually fade in over a duration
+ * @param priority the queue is ordered based on prioirty. The highest valued priority will be queued to play first
```

Comment:
similar to the fadeout phrasing, id say "priority in the queue..." as opposed to just explaining the fact that there is a queue